include/CL/sycl/buffer/detail/buffer.hpp

```diff
@@ -349,29 +349,12 @@ class buffer : public detail::buffer_base,
   }
 
 
-private:
-
-  /// Allocate uninitialized buffer memory
-  auto allocate_buffer(const range<Dimensions> &r) {
-    auto count = r.get_count();
-    // Allocate uninitialized memory
-    allocation = alloc.allocate(count);
-    return boost::multi_array_ref<value_type, Dimensions> { allocation, r };
-  }
-
-
-  /// Deallocate buffer memory if required
-  void deallocate_buffer() {
-    if (allocation)
-      alloc.deallocate(allocation, access.num_elements());
-  }
-
-
   /** Get a \c future to wait from inside the \c cl::sycl::buffer in
       case there is something to copy back to the host
 
       \return A \c future in the \c optional if there is something to
       wait for, otherwise an empty \c optional
+      \todo Make the function private again
```

Comment:
Instead of moving the code around it is simpler to understand history by just adding a
```
public:
```
at the right place.